### PR TITLE
Delegate address

### PR DESCRIPTION
**Pull request by descorp**
## Tested scenarios

**Fixed issue**:

### CODE REVIEW

File: android/src/main/java/com/adyenreactnativesdk/component/base/BaseModule.kt [hunk 1]
```diff
@@ -222,7 +225,11 @@ abstract class BaseModule(context: ReactApplicationContext?) : ReactContextBaseJ
         const val DID_PROVIDE = "didProvideCallback"
         const val DID_FAILED = "didFailCallback"
         const val DID_SUBMIT = "didSubmitCallback"
+        const val DID_UPDATE_ADDRESS = "didUpdateAddressCallback"
+        const val DID_CONFIRM_ADDRESS = "didConfirmAddressCallback"
+
```
Comment (Robert-SD) [inline android/src/main/java/com/adyenreactnativesdk/component/base/BaseModule.kt:230]:
```suggestion
```

File: android/src/main/java/com/adyenreactnativesdk/component/dropin/DropInModule.kt [hunk 2]
```diff
@@ -103,6 +118,43 @@ class DropInModule(context: ReactApplicationContext?) : BaseModule(context),
...
+    fun update(results: ReadableArray?) {
+        if (results == null) return
+        val listener = getService()
+        if (listener == null) {
+            sendErrorEvent(ModuleException.NoModuleListener(integration))
+            return
+        }
+
+        val jsonString = ReactNativeJson.convertArrayToJson(results).toString()
+        val addresses = Gson().fromJson(jsonString, Array<LookupAddress>::class.java)
```
Comment (Robert-SD) [inline android/src/main/java/com/adyenreactnativesdk/component/dropin/DropInModule.kt]: Would adding try/catch around .fromJson help for error handling?

File: ios/Configuration/CardConfigurationParser.swift [hunk 3]
```diff
@@ -75,17 +77,22 @@ public struct CardConfigurationParser {
...
 
-    // TODO: add installmentConfiguration: InstallmentConfiguration?
-
-    public var configuration: CardComponent.Configuration {
-        var storedConfiguration = StoredCardConfiguration()
-        storedConfiguration.showsSecurityCodeField = showsStoredSecurityCodeField
+    var storedCardConfiguration: StoredCardConfiguration {
+        var soredCardConfiguration = StoredCardConfiguration()
+        soredCardConfiguration.showsSecurityCodeField = showsStoredSecurityCodeField
+        return soredCardConfiguration
```
Comment (Robert-SD) [inline ios/Configuration/CardConfigurationParser.swift]:
```suggestion
        var storedCardConfiguration = StoredCardConfiguration()
        storedCardConfiguration.showsSecurityCodeField = showsStoredSecurityCodeField
        return storedCardConfiguration
```

File: ios/Configuration/CardConfigurationParser.swift [hunk 4]
```diff
@@ -75,17 +77,22 @@ public struct CardConfigurationParser {
...
 
+    var billingAddressConfiguration: BillingAddressConfiguration {
         var billingAddressConfiguration = BillingAddressConfiguration()
         billingAddressConfiguration.countryCodes = billingAddressCountryCodes
         billingAddressConfiguration.mode = addressVisibility
+        return billingAddressConfiguration
+    }
 
-        var soredCardConfiguration = StoredCardConfiguration()
+    // TODO: add installmentConfiguration: InstallmentConfiguration?
```
Comment (Robert-SD) [inline ios/Configuration/CardConfigurationParser.swift]: Do we have to remove the TODO?

File: src/wrappers/AdyenNativeComponentWrapper.tsx [hunk 5]
```diff
@@ -52,3 +45,5 @@ export class AdyenNativeComponentWrapper implements AdyenActionComponent {
     }
   }
 }
+
+
```
Comment (Robert-SD) [inline src/wrappers/AdyenNativeComponentWrapper.tsx]:
```suggestion
```

File: src/wrappers/getNativeComponent.tsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+    throw new Error(UNSUPPORTED_PAYMENT_METHOD_ERROR + typeName);
+  }
+
+  let nativeComponent: AdyenComponent & NativeModule;
+  if (ADDRESS_COMPONENTS.includes(typeName)) {
+    nativeComponent = new AdyenAddressLookupComponentWrapper({
+      nativeModule: AdyenDropIn,
+    });
+  } else {
+    // New iDEAL not treated as INSTANT on Android
```
Comment (Robert-SD) [inline src/wrappers/getNativeComponent.tsx:75]: Will this change in future?

File: ios/Configuration/CardConfigurationParser.swift (hunk 4 above)
Comment (descorp) [reply ios/Configuration/CardConfigurationParser.swift]: This still needs to be done 😅

File: android/src/main/java/com/adyenreactnativesdk/component/base/BaseModule.kt (hunk 1 above)
Comment (descorp) [reply android/src/main/java/com/adyenreactnativesdk/component/base/BaseModule.kt:230]: This one is ok

File: src/wrappers/getNativeComponent.tsx (hunk 6 above)
Comment (descorp) [reply src/wrappers/getNativeComponent.tsx:75]: I do hope so 🤷‍♂️

File: src/wrappers/AdyenNativeComponentWrapper.tsx (hunk 5 above)
Comment (descorp) [reply src/wrappers/AdyenNativeComponentWrapper.tsx]: Fixed [7261175](https://github.com/Adyen/adyen-react-native/pull/510/commits/72611758cd9b97b22aecd853f7dd130858ee9739)

File: ios/Configuration/CardConfigurationParser.swift (hunk 3 above)
Comment (descorp) [reply ios/Configuration/CardConfigurationParser.swift]: Fixed [7261175](https://github.com/Adyen/adyen-react-native/pull/510/commits/72611758cd9b97b22aecd853f7dd130858ee9739)

File: android/src/main/java/com/adyenreactnativesdk/component/dropin/DropInModule.kt (hunk 2 above)
Comment (descorp) [reply android/src/main/java/com/adyenreactnativesdk/component/dropin/DropInModule.kt]: Fixed [7261175](https://github.com/Adyen/adyen-react-native/pull/510/commits/72611758cd9b97b22aecd853f7dd130858ee9739)

**Review by Robert-SD — APPROVED** [review]

**Pull request merged**